test/LibraryManager.IntegrationTest/FileSaveRestoreTests.cs

```diff
@@ -0,0 +1,344 @@
+
+            Editor.Caret.MoveToExpression("\"version\": \"1.0\"");
+            Editor.Caret.MoveToEndOfLine();
+            Editor.KeyboardCommands.Enter();
+            Editor.KeyboardCommands.Type("\"defaultProvider\":");
+            LibmanTestsUtility.WaitForCompletionEntry(Editor, "cdnjs", caseInsensitive: true);
+            Editor.KeyboardCommands.Type("cdnjs");
+
+            Editor.Caret.MoveToExpression("\"libraries\"");
+            Editor.Caret.MoveDown(2);
+            Editor.KeyboardCommands.Type("\"library\":");
+            LibmanTestsUtility.WaitForCompletionEntry(Editor, "jquery", caseInsensitive: true, timeout: 5000);
+            Editor.KeyboardCommands.Type("jquery@3.3.1");
+            Editor.KeyboardCommands.Right();
+            Editor.KeyboardCommands.Type(",");
+            Editor.KeyboardCommands.Enter();
+            Editor.KeyboardCommands.Type("\"destination\":");
+            Editor.KeyboardCommands.Type("wwwroot/lib/jquery");
+
+            Editor.KeyboardCommands.Right();
+            Editor.KeyboardCommands.Type(",");
+            Editor.KeyboardCommands.Enter();
+            Editor.KeyboardCommands.Type("\"files\": [");
+            Editor.KeyboardCommands.Enter();
+            Editor.KeyboardCommands.Type("\"jquery.js\"");
+
+            _libManConfig.Save();
+            string cwd = Path.Combine(Path.GetDirectoryName(_webProject.FullPath), "wwwroot", "lib", "jquery");
+            LibmanTestsUtility.WaitForRestoredFile(cwd, "jquery.js", caseInsensitive: true);
+            LibmanTestsUtility.WaitForRestoredFileNotPresent(cwd, "jquery.min.js", caseInsensitive: true);
+        }
+
+        [TestMethod]
+        public void FileSaveRestore_Unpkg_AddNewLibraryWithSpecifingFiles()
+        {
+            _libManConfig.Open();
+
+            Editor.Caret.MoveToExpression("\"version\": \"1.0\"");
+            Editor.Caret.MoveToEndOfLine();
+            Editor.KeyboardCommands.Enter();
+            Editor.KeyboardCommands.Type("\"defaultProvider\":");
+            LibmanTestsUtility.WaitForCompletionEntry(Editor, "unpkg", caseInsensitive: true);
+            Editor.KeyboardCommands.Type("unpkg");
+
+            Editor.Caret.MoveToExpression("\"libraries\"");
+            Editor.Caret.MoveDown(2);
+            Editor.KeyboardCommands.Type("\"library\":");
+            Editor.KeyboardCommands.Type("jquery@3.3.1");
+            Editor.KeyboardCommands.Right();
+            Editor.KeyboardCommands.Type(",");
+            Editor.KeyboardCommands.Enter();
+            Editor.KeyboardCommands.Type("\"destination\":");
+            Editor.KeyboardCommands.Type("wwwroot/lib/jquery");
+
+            Editor.KeyboardCommands.Right();
+            Editor.KeyboardCommands.Type(",");
+            Editor.KeyboardCommands.Enter();
+            Editor.KeyboardCommands.Type("\"files\": [");
+            Editor.KeyboardCommands.Enter();
+            Editor.KeyboardCommands.Type("\"LICENSE.txt\"");
+
```

Comment:
A much more realistic user scenarios (i.e. much more worth protecting with a test) would be to get the jquery.min.js file.

Reply:
I see. I will do this check.

---
In reply to: [194552135](https://github.com/aspnet/LibraryManager/pull/113#discussion_r194552135) [](ancestors = 194552135)